src/include/sof/audio/buffer.h

```diff
@@ -154,7 +154,7 @@ struct buffer_cb_free {
 
 /* pipeline buffer creation and destruction */
 struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t align);
-struct comp_buffer *buffer_new(struct sof_ipc_buffer *desc);
+struct comp_buffer *buffer_new(const struct sof_ipc_buffer *desc);
```

Comment:
We are probably missing other const usage wrt IPC too.

Reply:
Probably, this one just came up when looking at your valgrind fixes.